src/layout/DwindleLayout.cpp

```diff
@@ -373,25 +406,18 @@ void CHyprDwindleLayout::onWindowCreatedTiling(CWindow* pWindow) {
     }
 
     // Update the children
-
-    if (NEWPARENT->size.x * *PWIDTHMULTIPLIER > NEWPARENT->size.y) {
-        // split left/right
-        OPENINGON->position = NEWPARENT->position;
-        OPENINGON->size     = Vector2D(NEWPARENT->size.x / 2.f, NEWPARENT->size.y);
-        PNODE->position     = Vector2D(NEWPARENT->position.x + NEWPARENT->size.x / 2.f, NEWPARENT->position.y);
-        PNODE->size         = Vector2D(NEWPARENT->size.x / 2.f, NEWPARENT->size.y);
+    if (verticalOverride) {
+        updateChildrenVertical(OPENINGON, PNODE, NEWPARENT);
+    } else if (NEWPARENT->size.x * *PWIDTHMULTIPLIER > NEWPARENT->size.y || horizontalOverride) {
+        updateChildrenHorizontal(OPENINGON, PNODE, NEWPARENT);
     } else {
-        // split top/bottom
-        OPENINGON->position = NEWPARENT->position;
-        OPENINGON->size     = Vector2D(NEWPARENT->size.x, NEWPARENT->size.y / 2.f);
-        PNODE->position     = Vector2D(NEWPARENT->position.x, NEWPARENT->position.y + NEWPARENT->size.y / 2.f);
-        PNODE->size         = Vector2D(NEWPARENT->size.x, NEWPARENT->size.y / 2.f);
+        updateChildrenVertical(OPENINGON, PNODE, NEWPARENT);
```

Comment:
de-duplicate!!! 2 outcomes 3 blocks

useless functions here, wont ever be used outside